main/src/ca/uwaterloo/flix/language/errors/RedundancyError.scala

```diff
@@ -408,4 +408,52 @@ object RedundancyError {
          |""".stripMargin
     })
   }
+
+  /**
+    * An error raised to indicate that a pure expression value was discarded.
+    *
+    * @param loc the location of the expression.
+    */
+  case class DiscardedPureValue(loc: SourceLocation) extends RedundancyError {
+    def summary: String = "Discarded a pure expression."
+
+    def message(formatter: Formatter): String = {
+      import formatter._
+      s"""${line(kind, source.name)}
+         |>> Discarded a pure expression: It should be impure.
+         |
+         |${code(loc, "discarded pure expression")}
+         |""".stripMargin
+    }
+
+    def explain(formatter: Formatter): Option[String] = None
+  }
+
+  /**
+    * An error raised to indicate that an impure unit value was discarded.
+    *
+    * @param loc the location of the expression.
+    */
+  case class DiscardedUnitValue(loc: SourceLocation) extends RedundancyError {
```

Comment:
What's the example here? I don't get it. Is it if you write `discard println("..")`?

Reply:
yes